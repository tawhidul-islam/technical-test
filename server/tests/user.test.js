const request = require("supertest");
require("dotenv").config();

const User = require("../models/User");

const baseUrl = process.env.BASE_URL;

beforeEach(async () => {
  await User.remove();
});

describe("users API", () => {
  it("POST /users --> create user", async () => {
    const response = await request(baseUrl)
      .post("/users")
      .send({
        name: "Jahid Hiron",
        email: "ayz@gmail.com",
        password: "jaHid1782?",
      })
      .expect(201);

    const user = await User.findOne({
      where: { userId: response._body.user.id },
    });

    expect(response._body).toMatchObject({
      user: {
        id: response._body.user.id,
        email: response._body.user.email,
      },
      token: response._body.token,
    });
    expect(user.password).not.toBe("jaHid1782?");
  });

  it("POST /users/signup-with-google --> ( signin && user_exist ) || signup", async () => {
    const response = await request(baseUrl)
      .post("/users/signup-with-google")
      .send({
        email: "namehiron.96@gmail.com",
        name: "Jahidul Islam Hiron",
        googleId: "115985087602801330728",
        avatar:
          "https://lh3.googleusercontent.com/a-/AOh14Gj3MBvKGdV0SG5v9LDxTi9uDIwltqCU9zjdb_TW5Q=s96-c",
      })
      .expect(201);

    expect(response._body).toMatchObject({
      user: {
        id: response._body.user.id,
        email: response._body.user.email,
      },
      token: response._body.token,
    });
  });
});
