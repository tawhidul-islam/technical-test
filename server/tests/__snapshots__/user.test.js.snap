// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users API POST /users --> create user 1`] = `
Object {
  "email": "axample@gmail.com",
  "id": "a05a9c3e-1b17-4dcb-944b-6f54791881d7",
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NiwiZW1haWwiOiJheGFtcGxlQGdtYWlsLmNvbSIsImlhdCI6MTY0NDU3NTA1MSwiZXhwIjoxNjQ0NjYxNDUxfQ.2w6gZU5OQQVXEPWM6LjK8Jo1AcNIJ71lKVA56WIJCAc",
}
`;
